test/integration/webhook/workload_test.go

```diff
@@ -194,5 +215,49 @@ var _ = ginkgo.Describe("Workload validating webhook", func() {
 			gomega.Expect(err).Should(gomega.HaveOccurred())
 			gomega.Expect(errors.IsForbidden(err)).Should(gomega.BeTrue(), "error: %v", err)
 		})
+
+		ginkgo.It("workload's priority should be mutable when referencing WorkloadPriorityClass", func() {
+			ginkgo.By("creating workload")
+			wl := testing.MakeWorkload("wl", ns.Name).Queue("lq").Request(corev1.ResourceCPU, "1").
+				PriorityClass("workload-priority-class").PriorityClassSource(constants.WorkloadPriorityClassSource).Priority(200).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+			gomega.Eventually(func() []metav1.Condition {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Status.Conditions
+			}, util.Timeout, util.Interval).ShouldNot(gomega.BeNil())
+			initialPriority := int32(200)
+			gomega.Expect(updatedQueueWorkload.Spec.Priority).To(gomega.Equal(&initialPriority))
+
+			ginkgo.By("Updating workload's priority")
+			updatedPriority := int32(150)
+			updatedQueueWorkload.Spec.Priority = &updatedPriority
+			gomega.Expect(k8sClient.Update(ctx, &updatedQueueWorkload)).To(gomega.Succeed())
+			gomega.Eventually(func() int32 {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(&updatedQueueWorkload), &finalQueueWorkload)).To(gomega.Succeed())
+				return *finalQueueWorkload.Spec.Priority
+			}, util.Timeout, util.Interval).Should(gomega.Equal(updatedPriority))
```

Comment:
```suggestion
				return finalQueueWorkload.Spec.Priority
			}, util.Timeout, util.Interval).Should(gomega.Equal(&updatedPriority))
```

I think there is a potential nil pointer error.